app/services/proofing/resolution/plugins/instant_verify_residential_address_plugin.rb

```diff
@@ -24,8 +24,36 @@ def call(
         end
 
         def proofer
-          @proofer ||=
-            if IdentityConfig.store.proofer_mock_fallback
+          @proofer ||= begin
+            # Historically, proofer_mock_fallback has controlled whether we
+            # use mock implementations of the Resolution and Address proofers
+            # (true = use mock, false = don't use mock).
+            # We are transitioning to a place where we will have separate
+            # configs for both. For the time being, we want to keep support for
+            # proofer_mock_fallback here. This can be removed after this code
+            # has been deployed and configs have been updated in all relevant
+            # environments.
+
+            old_config_says_mock = IdentityConfig.store.proofer_mock_fallback
+            old_config_says_iv = !old_config_says_mock
+            new_config_says_mock =
+              IdentityConfig.store.idv_resolution_default_vendor == :mock
+            new_config_says_iv =
+              IdentityConfig.store.idv_resolution_default_vendor == :instant_verify
+
+            proofer_type =
+              if new_config_says_mock && old_config_says_iv
+                # This will be the case immediately after deployment, when
+                # environment configs have not been updated. We need to
+                # fall back to the old config here.
```

Comment:
Very minor: the examples in the tests say "the new setting takes precedence," but here the comment says we "fall back to the old config" which suggests it's the other way around. They're not discussing the same thing and the behavior is actually consistent, but if you see an easy way to make the wording consistent, it might further prevent confusion for anyone looking at this first thing in the morning while under-caffeinated. (Hypothetically speaking.)

Reply:
I think this is a case of bad spec name. I changed it to `creates an Instant Verify proofer because the new setting takes precedence over the old one when the old one is set to its default value`. The idea is that, when proofer_mock_fallback is false, that is an indication that it has (probably) not been actively configured. If the new setting is set to a non-default value, we should use that instead.